README.md

```diff
@@ -174,3 +188,21 @@ In order to gain access to the the s3 bucket we will also need to provide our ac
 ```bash
 aws ec2 describe-security-groups --group-names 'Default' --query 'SecurityGroups[0].OwnerId' --output text
 ```
+
+## IP Blocklist
+The ability to restrict access to Nubis enviroments based on IP address is a basic security requirement. While Networ ACLs are used for the actual blocking, that administrator script runs from the nat instance and therefore is documented here.
+
+### Implementation
+Within the [nat](https://github.com/nubisproject/nubis-nat) instances we have implemented an IP blocklist [shell script](https://github.com/nubisproject/nubis-nat/blob/master/nubis/puppet/files/vpc-blocklist) that updates the default [Network Access Controll List](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_ACLs.html) (NACL) for the [Vitrual Private Cloud](https://aws.amazon.com/vpc/) (VPC). The script runs on a [cron job](https://github.com/nubisproject/nubis-nat/blob/master/nubis/puppet/04_blocklist.pp) and simply updates the default NACL for the VPC. Necessary permissions can be seen [here](link to security policy document).
+
+### Blocklist File
+The Mozilla Infosec team maintains a blocklist for AWS located [here](https://s3.amazonaws.com/mozilla_infosec_blocklist/blocklist). This list is auto generated from their security monitoring systems with the necessary resolution. Their full list often contains far more that 18 entries (see [Limitations](#limitations) below). For that reason they make decisions about what IPs to include based on their own priority.
+
+The current IP block list is not signed. We have requested that infosec implement signing of the generated file. The Nubis project already supports verification of signature and will enable the feature once signing is implemented.
+
+Alternate blocklists can be used by updating the deployment variable [Not yet implemented, link once complete).
+
+### Limitations
+There are a number of limitations that exist in AWS. There can only be [one NACL](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Appendix_Limits.html#vpc-limits-nacls) attached to a VPC at a time. A NACL can have up to [20 entries](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Appendix_Limits.html#vpc-limits-nacls). Of those 20 entries two are used by default and cannot be removed, a [??? entry](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_ACLs.html#default-network-acl) and a final [0.0.0.0/0](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_ACLs.html#default-network-acl) allow. This leaves 18 usable entries for block listing. This is a "soft" limit and can be raised to a "hard" limit of 40, however the [documentation](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Appendix_Limits.html#vpc-limits-nacls) suggests that this can lead to significant network slowdowns. For that reason we have decided to leave the number of entries at the default of 20.
```

Comment:
There is basically an ALLOW ALL rule at the top and a DENY ALL at the bottom, beats me why